### PR TITLE
SONAR-6988 Add cross project duplication info in batch report

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-6988 Add cross project duplication info in batch report

Add duplication blocks
Add isCrossProjectDuplicationActivated boolen in metada

### CODE REVIEW

File: sonar-batch-protocol/src/main/protobuf/batch_report.proto [hunk 1]
```diff
@@ -144,11 +145,21 @@ message Duplication {
   repeated Duplicate duplicate = 2;
 }
 
+// Used for cross project duplication
```
Comment (henryju) [inline sonar-batch-protocol/src/main/protobuf/batch_report.proto]: As mentioned by @sns-seb  you can remove other_file_key line 139

**Comment by henryju** [comment]
LGTM (but build is failing)

File: sonar-batch-protocol/src/main/protobuf/batch_report.proto (hunk 1 above)
Comment (julienlancelot) [inline sonar-batch-protocol/src/main/protobuf/batch_report.proto]: Indeed !

Comment (julienlancelot) [inline sonar-batch-protocol/src/main/protobuf/batch_report.proto]: In fact seb will do it on his branch as he's working on this.

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protocol' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'input' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/input) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'output' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/output) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'ws' package (org.sonarsource.sonarqube:sonar-ws:target/generated-sources/protobuf/org/sonarqube/ws) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: sonar-batch-protocol/src/main/protobuf/batch_report.proto (hunk 1 above)
Comment (sns-seb) [inline sonar-batch-protocol/src/main/protobuf/batch_report.proto]: I wasn't aware of that, but ok :)

Comment (julienlancelot) [inline sonar-batch-protocol/src/main/protobuf/batch_report.proto]: In fact I've tried to remove it but it's used by PersistDuplicationsStep -> As you'll work on this class, I think it's more simpler for you to do the change.

Comment (sns-seb) [inline sonar-batch-protocol/src/main/protobuf/batch_report.proto]: I agree

**Pull request merged**